AirCasting/SDSync/ViewsFlow/ClearingSDCard/ClearingSDCardViewModel.swift

```diff
@@ -1,18 +1,95 @@
+    var isClearingCompleted: Published<Bool>.Publisher { $isClearingCompletedValue }
 
     var isSDClearProcess: Bool
+    private let userAuthenticationSession: UserAuthenticationSession
+    private let peripheral: CBPeripheral
+    private let airBeamConnectionController: AirBeamConnectionController
+    private let sdSyncController: SDSyncController
+    private var error = ClearingSDCardError.undefined
+    @Published private var isClearingCompletedValue: Bool = false
+    @Published private var shouldDismissValue: Bool = false
     @Published var presentNextScreen: Bool = false
     
-    init(isSDClearProcess: Bool) {
+    
+    init(isSDClearProcess: Bool, userAuthenticationSession: UserAuthenticationSession, peripheral: CBPeripheral, airBeamConnectionController: AirBeamConnectionController, sdSyncController: SDSyncController) {
         self.isSDClearProcess = isSDClearProcess
+        self.userAuthenticationSession = userAuthenticationSession
+        self.peripheral = peripheral
+        self.airBeamConnectionController = airBeamConnectionController
+        self.sdSyncController = sdSyncController
+    }
+    
+    func clearSDCard() {
+        self.airBeamConnectionController.connectToAirBeam(peripheral: peripheral) { success in
+            guard success else {
+                DispatchQueue.main.async {
+                    self.isClearingCompletedValue = false
+                    self.shouldDismissValue = true
+                    self.error = ClearingSDCardError.noConnection
+                }
+                return
+            }
+            self.sendClearConfig()
+            self.sdSyncController.clearSDCard(self.peripheral) { result in
+                DispatchQueue.main.async {
+                    self.isClearingCompletedValue = result
+                    self.shouldDismissValue = !result
+                    if !result {
+                        self.error = ClearingSDCardError.noClearing
+                    }
+                }
+            }
+        }
+    }
+    
+    func getAlertTitle() -> String {
+        switch error {
+        case .undefined:
+            return Strings.ClearingSDCardView.failedClearingAlertTitle
+        case .noConnection:
+            return Strings.AirBeamConnector.connectionTimeoutTitle
+        case .noClearing:
+            return Strings.ClearingSDCardView.failedClearingAlertTitle
+        }
+    }
+    
+    func getAlertMessage() -> String {
+        switch error {
+        case .undefined:
+            return Strings.ClearingSDCardView.failedClearingAlertMessage
+        case .noConnection:
+            return Strings.AirBeamConnector.connectionTimeoutDescription
+        case .noClearing:
+            return Strings.ClearingSDCardView.failedClearingAlertMessage
+        }
+    }
+    
+    private func sendClearConfig() {
+        let configurator = AirBeam3Configurator(userAuthenticationSession: self.userAuthenticationSession,
```

Comment:
I'd change it so that `BluetoothSDCardAirBeamServices` sends those configurations. Both for download and for erase. We want only super high level stuff here, not some obscure configuration sending. Would that be possible?